src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -336,21 +323,18 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
      * Turn all our kernelConnections that we know about into registered NotebookControllers
      */
     private async loadNotebookControllersImpl(
-        options:
-            | { listLocalNonPythonKernels: boolean; useCache: 'useCache' | 'ignoreCache' }
-            | { listRemoteKernels: boolean }
+        options: { useCache: 'useCache' | 'ignoreCache' } | { listRemoteKernels: boolean }
     ): Promise<void> {
         const cancelToken = new CancellationTokenSource();
         this.wasPythonInstalledWhenFetchingControllers = this.extensionChecker.isPythonExtensionInstalled;
         let connections: KernelConnectionMetadata[] = [];
-        if ('listLocalNonPythonKernels' in options) {
-            connections = await (options.listLocalNonPythonKernels
-                ? this.localKernelFinder.listNonPythonKernels(cancelToken.token, options.useCache)
-                : this.localKernelFinder.listKernels(undefined, cancelToken.token, options.useCache)
-            ).catch((ex) => {
-                traceError('Failed to get local kernel connections', ex);
-                return [] as KernelConnectionMetadata[];
-            });
+        if ('useCache' in options) {
```

Comment:
you could switch this if statement to make it more clear what's going on: if (listRemoteKernels)